File: selfstudy/python/cities_sortdesc.py
'''
task 180:
Cities - SortDesc
You need to sort the values in descending order for city populations in thousands
'''

def sort_down_cities(populations):
    return sorted(populations, reverse=True)

# Tests
if __name__ == '__main__':
    tests = [
        ([10, 20, 30], sort_down_cities([10, 20, 30])),
        ([50, 60, 70], sort_down_cities([50, 60, 70])),
        ([1, 2, 3], sort_down_cities([1, 2, 3])),
        ([100, 200], sort_down_cities([100, 200])),
        ([], sort_down_cities([]))
    ]
    import sys
    if 'pytest' in sys.argv:
        # pytest structure
        pass
    else:
        for idx, (inp, exp) in enumerate(tests):
            res = sort_down_cities(inp)
            print(f"Test {idx+1}:", "PASS" if res == exp else "FAIL")

# done

File: selfstudy/golang/cities_sortdesc.go
/*
task 280:
Cities - SortDesc
You need to sort the values in descending order for city populations in thousands
*/

package main

import (
	"fmt"
	"reflect"
)

func Sort_downCities(populations []int) []int {
	res := make([]int, len(populations))
	copy(res, populations)
	for i := 0; i < len(res); i++ {
		for j := i+1; j < len(res); j++ {
			if res[i] < res[j] { res[i], res[j] = res[j], res[i] }
		}
	}
	return res
}

func main() {
	tests := []struct {
		input []int
		expected []int
	}{
		{[]int{10, 20, 30}, Sort_downCities([]int{10, 20, 30})},
		{[]int{50, 60, 70}, Sort_downCities([]int{50, 60, 70})},
		{[]int{1, 2, 3}, Sort_downCities([]int{1, 2, 3})},
		{[]int{100, 200}, Sort_downCities([]int{100, 200})},
		{[]int{}, Sort_downCities([]int{})},
	}
	
	for idx, t := range tests {
		res := Sort_downCities(t.input)
		status := "FAIL"
		if reflect.DeepEqual(res, t.expected) {
			status = "PASS"
		}
		fmt.Printf("Test %d: %s\n", idx+1, status)
	}
}

// works i think
